targeted/json_csrf_poc_generator.js

```diff
@@ -0,0 +1,97 @@
+//csrf poc generater supporting json csrf
+//also supports multipart/form-data .
+//it will copy the results to clipboard and print them to the zap script console
+// released under the Apache v2.0 licence.
+//You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//Author : @haseebeqx
+
+function invokeWith(msg) {
+	string = "<!DOCTYPE html>\n";
+	string += "<head>\n <title>CSRF POC</title>\n</head>";
+	string += "\n<body>";
+	if(msg.getRequestHeader().getMethod()== "POST"){
+	body = msg.getRequestBody().toString();
+	body = body.trim();
+	if(isJson(body))
+		string += "\n<form action=\""+ msg.getRequestHeader().getURI().toString() +"\" id=\"formid\" method=\"post\" enctype=\"text/plain\">" ;
+	else{
+		if(ismultipart(msg.getRequestHeader()))
+			string += "\n<form action=\""+ msg.getRequestHeader().getURI().toString() +"\" id=\"formid\" enctype=\"multipart/form-data\" method=\"post\">";
+		else
+			string += "\n<form action=\""+ msg.getRequestHeader().getURI().toString() +"\" id=\"formid\" method=\"post\">";
+	}
+	if(!isJson(body)){	
+	if(ismultipart(msg.getRequestHeader())){
+		type 	=  msg.getRequestHeader().getHeader(org.parosproxy.paros.network.HttpRequestHeader.CONTENT_TYPE);
```

Comment:
Probably better to use "HttpHeader" instead of "HttpRequestHeader", the latter does not work in Nashorn.